awx/main/management/commands/cleanup_jobs.py

```diff
@@ -58,27 +62,52 @@ def add_arguments(self, parser):
                             help='Remove workflow jobs')
 
     def cleanup_jobs(self):
-        #jobs_qs = Job.objects.exclude(status__in=('pending', 'running'))
-        #jobs_qs = jobs_qs.filter(created__lte=self.cutoff)
-        skipped, deleted = 0, 0
-        jobs = Job.objects.filter(created__lt=self.cutoff)
-        for job in jobs.iterator():
-            job_display = '"%s" (%d host summaries, %d events)' % \
-                          (str(job),
-                           job.job_host_summaries.count(), job.job_events.count())
-            if job.status in ('pending', 'waiting', 'running'):
-                action_text = 'would skip' if self.dry_run else 'skipping'
-                self.logger.debug('%s %s job %s', action_text, job.status, job_display)
-                skipped += 1
+        skipped, num_deleted = 0, 0
+
+        num_to_delete = Job.objects.filter(created__lt=self.cutoff).count()
+
+        batch_size = 10000
+
+        self.logger.info("deleting total of %d jobs", num_to_delete)
+        while True:
+            pk_list = Job.objects.filter(created__lt=self.cutoff)[0:batch_size].values_list('pk')
+            qs_batch = Job.objects.filter(pk__in=pk_list)
+            if not self.dry_run:
+                just_deleted, _ = qs_batch.delete()
             else:
-                action_text = 'would delete' if self.dry_run else 'deleting'
-                self.logger.info('%s %s', action_text, job_display)
-                if not self.dry_run:
-                    job.delete()
-                deleted += 1
+                just_deleted = qs_batch.count()
+            num_deleted += just_deleted
+            self.logger.info('deleted %d jobs', num_deleted)
+            if just_deleted == 0:
+                break
 
         skipped += Job.objects.filter(created__gte=self.cutoff).count()
-        return skipped, deleted
+        return skipped, num_deleted
+
+    def cleanup_jobs_fast(self):
+        skipped, num_deleted = 0, 0
+
+        num_to_delete = Job.objects.filter(created__lt=self.cutoff).count()
+
+        batch_size = 1000000
+
+        self.logger.info("deleting total of %d jobs", num_to_delete)
+        while True:
+            collector = AWXCollector('default', self.logger)
+            pk_list = Job.objects.filter(created__lt=self.cutoff)[0:batch_size].values_list('pk')
+            qs_batch = Job.objects.filter(pk__in=pk_list)
+            num_deleted += qs_batch.count()
```

Comment:
Probably can update this as well to remove the `.count()` (instead, just add the `just_deleted` on each loop, like the function above).

Reply:
thanks, fixed!